clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -2445,6 +2445,46 @@ static bool isTypeSubstitutable(Qualifiers Quals, const Type *Ty,
   return true;
 }
 
+namespace {
+using DeclContextDesc = std::pair<clang::Decl::Kind, StringRef>;
+bool matchQualifiedTypeName(const QualType &Ty,
+                            ArrayRef<DeclContextDesc> Scopes) {
+  // The idea: check the declaration context chain starting from the type
+  // itself. At each step check the context is of expected kind
+  // (namespace) and name.
+  const CXXRecordDecl *RecTy = Ty->getAsCXXRecordDecl();
+
+  if (!RecTy)
```

Comment:
Can this be an assert instead? I'd hate for us to think this works for other things, then it fails.

Reply:
In this file I call this function with `TagType`, which might represent both Record and Enum. So, I would prefer `if` instead